### PR TITLE
qat: copy annotations

### PR DESCRIPTION
**Pull request by bart0sh**
- Copied annotations from the QAT device plugin CR to the QAT deployment
- Tested with envtest

Fixes: #645

### CODE REVIEW

File: deployments/operator/device/qat/qat.yaml [hunk 1]
```diff
@@ -5,6 +5,9 @@ metadata:
   namespace: inteldeviceplugins-system
 spec:
   template:
+    metadata:
+      annotations:
+        container.apparmor.security.beta.kubernetes.io/intel-qat-plugin: unconfined
```
Comment (mythi) [inline deployments/operator/device/qat/qat.yaml:10]: we don't set this on by default with kustomize but it's provided as a sample overlay. I'm thinking maybe se should keep this here as an example but comment it out.

File: pkg/controllers/qat/controller.go [hunk 2]
```diff
@@ -78,13 +78,15 @@ func (c *controller) GetTotalObjectCount(ctx context.Context, clnt client.Client
 func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	devicePlugin := rawObj.(*devicepluginv1.QatDevicePlugin)
 	yes := true
+	pluginAnnotations := devicePlugin.ObjectMeta.DeepCopy().Annotations
```
Comment (mythi) [inline pkg/controllers/qat/controller.go:81]: `UpdateDaemonSet` needs changing too: if the user edits existing `QatDevicePlugin` the annotations are not added/removed

File: test/envtest/qatdeviceplugin_controller_test.go [hunk 3]
```diff
@@ -46,9 +46,14 @@ var _ = Describe("QatDevicePlugin Controller", func() {
...
 
+			annotations := map[string]string{
+				"container.apparmor.security.beta.kubernetes.io/intel-qat-plugin": "unconfined",
+			}
+
 			toCreate := &devicepluginv1.QatDevicePlugin{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: key.Name,
+					Name:        key.Name,
+					Annotations: annotations,
```
Comment (mythi) [inline test/envtest/qatdeviceplugin_controller_test.go:56]: test with and without, e.g., by adding them like this but then updating the CR and checking they're not there anymore?

**Review by mythi — COMMENTED** [review]
thanks @bart0sh, this is what I had in mind. I added a couple of thoughts.

File: deployments/operator/device/qat/qat.yaml (hunk 1 above)
Comment (bart0sh) [reply deployments/operator/device/qat/qat.yaml]: done

File: pkg/controllers/qat/controller.go (hunk 2 above)
Comment (bart0sh) [reply pkg/controllers/qat/controller.go:81]: done

File: test/envtest/qatdeviceplugin_controller_test.go (hunk 3 above)
Comment (bart0sh) [reply test/envtest/qatdeviceplugin_controller_test.go:56]: done

**Comment by bart0sh** [comment]
@mythi thank you for the review. updated the PR according to your comments.

File: deployments/operator/device/qat/qat.yaml [hunk 4]
```diff
@@ -5,6 +5,13 @@ metadata:
   namespace: inteldeviceplugins-system
 spec:
   template:
+    # example apparmor annotation
```
Comment (mythi) [inline deployments/operator/device/qat/qat.yaml:8]: just realized these are in a wrong place. they belong to `deployments/operator/samples/`

Comment (bart0sh) [reply deployments/operator/device/qat/qat.yaml]: fixed

**Review by mythi — APPROVED** [review]
tested and works ok!

**Review by rojkov — APPROVED** [review]

**Pull request merged**